indexer-app/src/cmr/indexer/data/concepts/association_util.clj

```diff
@@ -2,19 +2,26 @@
   "Contains functions to parse and convert generic associations to a map structure 
   for each generic concept type."
   (:require
+   [clojure.set :as s]
    [cmr.common.concepts :as cc]))
 
 (defn allow-group-by-concept-type
   "Creates a temporary structure so that we can group the concept ids by concept type."
-  [concept-id]
-  {:concept-type (cc/concept-id->type concept-id)
-   :concept-id concept-id})
+  [association concept-id]
+  (let [asn (if (= concept-id (:source-concept-identifier association))
+              (-> association
+                  (dissoc :source-concept-identifier)
+                  (s/rename-keys {:associated-concept-id :concept-id}))
+              (-> association
+                  (dissoc :associated-concept-id)
+                  (s/rename-keys {:source-concept-identifier :concept-id})))]
+    (assoc asn :concept-type (cc/concept-id->type (:concept-id asn)))))
```

Comment:
I think you might want to rename the source/associated revision-ids to revision-id.  Then you could make use of it in the search result to include it in the association-details.

Reply:
I made the change.